diff --git a/plugins/layer_types/openlayers_layer_type_geojson.js b/plugins/layer_types/openlayers_layer_type_geojson.js
index f694d26..d2fde7f 100644
--- a/plugins/layer_types/openlayers_layer_type_geojson.js
+++ b/plugins/layer_types/openlayers_layer_type_geojson.js
@@ -36,6 +36,7 @@ Drupal.openlayers.layer.geojson = function(title, map, options) {
     // Add features, if needed
     if (features) {
       layer.addFeatures(features);
+      layer.events.triggerEvent('loadend');
     }
   }
   else if (options.useBBOX) {
@@ -56,6 +57,7 @@ Drupal.openlayers.layer.geojson = function(title, map, options) {
       // Add features, if needed
       if (features) {
         layer.addFeatures(features);
+        layer.events.triggerEvent('loadend');
       }
     });
   }
diff --git a/plugins/layer_types/openlayers_layer_type_kml.js b/plugins/layer_types/openlayers_layer_type_kml.js
index d690cbc..0651d4e 100644
--- a/plugins/layer_types/openlayers_layer_type_kml.js
+++ b/plugins/layer_types/openlayers_layer_type_kml.js
@@ -27,6 +27,7 @@ Drupal.openlayers.layer.kml = function(title, map, options) {
     // Add features, if needed
     if (features) {
       layer.addFeatures(features);
+      layer.events.triggerEvent('loadend');
     }
   });
 
