diff --git a/includes/webform.report.inc b/includes/webform.report.inc
index 6ada86a..8fc1d03 100644
--- a/includes/webform.report.inc
+++ b/includes/webform.report.inc
@@ -807,6 +807,7 @@ function webform_results_export($node, $format = 'delimited', $options = array()
   $export_info['file_name'] = $file_name;
   $export_info['exporter'] = $exporter;
   $export_info['row_count'] = $row_count;
+  $export_info['last_sid'] = $sid;
 
   return $export_info;
 }
@@ -829,7 +830,7 @@ function webform_results_download($node, $export_info) {
   @unlink($export_info['file_name']);  // Clean up, the @ makes it silent.
 
   // Update user last downloaded sid if required.
-  if ($options['range_type'] != 'range' && $row_count > 0) {
+  if ($export_info['options']['range_type'] != 'range' && $export_info['row_count'] > 0) {
     // Delete existing record.
     db_delete('webform_last_download')
       ->condition('nid', $node->nid)
@@ -840,7 +841,7 @@ function webform_results_download($node, $export_info) {
       ->fields(array(
         'nid' => $node->nid,
         'uid' => $user->uid,
-        'sid' => $sid,
+        'sid' => $export_info['last_sid'],
         'requested' => REQUEST_TIME,
       ))
       ->execute();
