diff --git a/modules/views.page_title.inc b/modules/views.page_title.inc
index b4c9c98..b51eeb4 100644
--- a/modules/views.page_title.inc
+++ b/modules/views.page_title.inc
@@ -45,8 +45,10 @@ function views_page_title_pattern_alter(&$pattern, &$types) {
       }
     }
     // This is a view with no args provided, or the specific arg has no title - lets use the base display title
-    elseif($view->display->display_plugin == 'page_with_page_title') {
-      $pattern = $view->display->handler->get_option('page_title_pattern');
+    elseif(isset($view->display->display_plugin)) {
+      if($view->display->display_plugin == 'page_with_page_title') {
+        $pattern = $view->display_handler->get_option('page_title_pattern');
+      }
     }
   }
 }
